### PR TITLE
chore: add prettier precommit hook

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

This pull request adds a pre-commit hook to format files with `prettier`. The advantages of this is that there is no longer a reliance on developers setting up their environment to run `prettier`. Regardless of their setup, the staged files will format on commit if they are not already formatted.

Note: Running `yarn postinstall` is now a new requirement when cloning the repo (added to README). Typically, the `postinstall` is run automatically after `install` but there is an issue with using `yarn` workspaces: https://yarnpkg.com/advanced/error-codes#yn0006---soft_link_build

### CODE REVIEW

File: gateway/packages/server/src/mocks/jwt-service.mock.ts [hunk 1]
```diff
@@ -1,7 +1,5 @@
 export class JwtServiceMock {
```
Comment (jpangelle) [inline gateway/packages/server/src/mocks/jwt-service.mock.ts:1]: auto `prettier` changes

File: gateway/packages/server/src/mocks/mailer-service.mock.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
 import { ISendMailOptions } from '@nestjs-modules/mailer/dist/interfaces/send-mail-options.interface'
```
Comment (jpangelle) [inline gateway/packages/server/src/mocks/mailer-service.mock.ts:1]: auto `prettier` changes

File: gateway/packages/ui/src/schemas/SchemaList.tsx [hunk 3]
```diff
@@ -72,18 +72,16 @@ const formModePropMapping = {
 }
```
Comment (jpangelle) [inline gateway/packages/ui/src/schemas/SchemaList.tsx:72]: auto `prettier` changes

File: onboarding/api/src/repos/address.repo.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export class AddressRepo {
```
Comment (jpangelle) [inline onboarding/api/src/repos/address.repo.ts:30]: auto `prettier` changes

File: onboarding/api/src/repos/kyc.repo.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export class KycRepo {
     `
```
Comment (jpangelle) [inline onboarding/api/src/repos/kyc.repo.ts:39]: auto `prettier` changes

File: onboarding/api/src/repos/user.repo.ts [hunk 6]
```diff
@@ -38,7 +38,7 @@ export class UserRepo {
       group by users.id, kyc.user_id, kyc.provider, kyc.provider_account_id, kyc.digest, kyc.created_at, kyc.status, kyc.usa_tax_resident, kyc.accredited
```
Comment (jpangelle) [inline onboarding/api/src/repos/user.repo.ts:38]: auto `prettier` changes

File: tinlake-ui/services/apollo/index.ts [hunk 7]
```diff
@@ -583,7 +583,7 @@ function toTinlakeLoans(loans: any[]): { data: Loan[] } {
```
Comment (jpangelle) [inline tinlake-ui/services/apollo/index.ts:583]: auto `prettier` changes

File: tinlake-ui/services/tinlake/index.ts [hunk 8]
```diff
@@ -17,7 +17,7 @@ export function initTinlake({
     // const overrides = config.network === 'Kovan' ? { gasLimit: config.gasLimit } : {}
```
Comment (jpangelle) [inline tinlake-ui/services/tinlake/index.ts:17]: auto `prettier` changes

**Review by offerijns — APPROVED** [review]

**Pull request merged**